### ISSUE TITLE
4.2.1234 Film Simulation Crash

### ISSUE THREAD

**Issue by torridgristle** [Beep6581/RawTherapee#3499]
This was working fine with an older version of RawTherapee from a few months ago and became an issue the moment I updated today. I didn't modify the HaldCLUT directory at all between the last time I used Film Simulation in the old RawTherapee and the first time I tried to use Film Simulation in RawTherapee 4.2.1234

RawTherapee would crash and Windows would show me the "rawtherapee.exe has stopped responding" message if I clicked on the Film Simulation dropdown menu. No text would load, it would just freeze for a second and then crash.

I got it to stop crashing by removing my Mine sub-directory from the HaldCLUT directory, but keeping the single HaldCLUT from the sub-directory in the root of the HaldLUT directory lets it load fine so it's probably not an issue with the file. I had five other sub-directories in the HaldCLUT directory, one with five and one with six sub-directories of their own.

It would crash if I had another sub-directory named anything, like Mine or My Own or "a", with a file inside of it. I tested with the attached file and Negative.png renamed to foo.png

![fuji super g 100 fe ammonium edta](https://cloud.githubusercontent.com/assets/23273466/20028090/6ad1e734-a2fe-11e6-9db2-efcf6ceb02a8.png)

**Comment by Beep6581** [comment]
Could you count separately how many files total and folders total you have in your Hald CLUT folder? I don't know how to get those two separate numbers in Windows, in Linux I would:
`find HaldCLUT_root_folder -type d | wc -l` for a total folder count, and
`find HaldCLUT_root_folder -type f | wc -l` for a total file count.

**Comment by torridgristle** [comment]
There's 330 files including the attached file from earlier, 21 folders, and one of the files is the README.txt Currently RawTherapee works fine.

Added Mine, moved attached from root HaldCLUT to Mine, RawTherapee crashes.

Moved a random file out of a different subdirectory and put it into the root HaldCLUT, attached still inside Mine, RawTherapee crashes.

Moved the previous random file out of the root HaldCLUT, placed it on my Desktop outside of the install directory of RawTherapee, 329 files and 21 folders, RawTherapee crashes.

Put the previous random file back into the subdirectory it came from, moved attached out of Mine and back into the root HaldCLUT so that Mine is empty, swapped names of the subdirectories Mine and Color (a subdiretory filled with 194 files and 5 folders) so that now Color is the empty directory and Mine is filled with the 194 files and 5 folders, RawTherapee works.

Restored proper names for Color and Mine, put Negative.png from the root HaldCLUT into Mine, attached is still in root HaldCLUT, RawTherapee crashes.

**Comment by Beep6581** [comment]
Related: cc585058abdb003a338865a8807dc3275fc1b110

**Comment by Floessie** [comment]
@Beep6581 If you point me to an easy reproduction (a bash line to create the needed files and folders or something), I'll have a look at it. A backtrace would also probably do.

**Comment by Beep6581** [comment]
@Floessie if I understood your request correctly, this script will copy a chosen file into 10 folders 100 times in each:
`for d in {1..10}; do mkdir $d; for f in {1..100}; do cp ~/profiles/HaldCLUT/Hald_CLUT_Identity_12.tif $d/${f}.tif; done; done`

**Comment by Floessie** [comment]
If it's not reproducible under Linux, I cannot help. :( And this is no OOM problem?

**Comment by Beep6581** [comment]
@torridgristle can you summarize, what are the minimum required steps to reproduce?
Can you run the debug build to get a stack backtrace? Sounds complicated but its quite simple: http://rawpedia.rawtherapee.com/How_to_write_useful_bug_reports
http://rawtherapee.com/releases_head/windows/RawTherapee_WinVista_64_Gtk3_Debug_4.2.1234.zip

**Comment by torridgristle** [comment]
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Windows\System32>cd C:\Program Files\RawTherapee

C:\Program Files\RawTherapee>gdb.exe rawtherapee.exe
GNU gdb (GDB) 7.9
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-w64-mingw32".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
This binary was built by Equation Solution <http://www.Equation.com>...
Reading symbols from rawtherapee.exe...done.
(gdb) r
Starting program: C:\Program Files\RawTherapee\rawtherapee.exe
[New Thread 2548.0x1828]
[New Thread 2548.0x1b9c]
[New Thread 2548.0x20e4]
[New Thread 2548.0x2374]
[New Thread 2548.0x20cc]
[New Thread 2548.0x1e5c]
[New Thread 2548.0x2618]
[New Thread 2548.0x2ba8]
[New Thread 2548.0x19b4]
[New Thread 2548.0x1ff0]
[New Thread 2548.0x2eb8]
[New Thread 2548.0x2b18]
[New Thread 2548.0x18bc]
[Thread 2548.0x2b18 exited with code 0]
[Thread 2548.0x18bc exited with code 0]
[New Thread 2548.0x207c]
[New Thread 2548.0x2d40]
[New Thread 2548.0x147c]
warning: `C:\PROGRA~1\AVASTS~1\Avast\1033\Base.dll': Shared library architecture
 i386 is not compatible with target architecture i386:x86-64.
[New Thread 2548.0x224c]
[Thread 2548.0x224c exited with code 0]
[New Thread 2548.0x2948]
[Thread 2548.0x2948 exited with code 0]
[New Thread 2548.0xe2c]
[Thread 2548.0x2d40 exited with code 0]
[New Thread 2548.0x1740]
[New Thread 2548.0x174]
[Thread 2548.0xe2c exited with code 0]
[New Thread 2548.0x27a8]
[Thread 2548.0x1740 exited with code 0]
[New Thread 2548.0x29d0]
[Thread 2548.0x29d0 exited with code 0]
[New Thread 2548.0xf24]
[Thread 2548.0x27a8 exited with code 0]

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory

(rawtherapee.exe:2548): Gtk-WARNING **: Failed to fetch network locations: Error
 opening directory 'C:\Program Files\RawTherapee\network:': No such file or dire
ctory
[New Thread 2548.0xcd8]
[New Thread 2548.0x2268]
[New Thread 2548.0x21f0]
[New Thread 2548.0x1c78]
[Thread 2548.0xf24 exited with code 0]
[Thread 2548.0x2268 exited with code 0]
[Thread 2548.0xcd8 exited with code 0]
[Thread 2548.0x1c78 exited with code 0]
[New Thread 2548.0x1f28]
[New Thread 2548.0x2df8]
[New Thread 2548.0xe28]
[New Thread 2548.0x2334]
[New Thread 2548.0x26b8]
[New Thread 2548.0x2fbc]
[New Thread 2548.0x1b34]
[New Thread 2548.0x1810]
[New Thread 2548.0x22c4]
[New Thread 2548.0x1be0]
[New Thread 2548.0x139c]
[New Thread 2548.0x1f94]
[New Thread 2548.0x2c68]
[New Thread 2548.0x26f0]
[New Thread 2548.0x2dc8]

Program received signal SIGSEGV, Segmentation fault.
0x0000000005598b57 in ?? () from C:\Program Files\RawTherapee\libgtk-3-0.dll
(gdb)

**Comment by Beep6581** [comment]
The important part is missing: "bt full". You can leave everything before "Program received signal SIGSEGV, Segmentation fault" out.

**Comment by torridgristle** [comment]
[rawtherapeelog.txt](https://github.com/Beep6581/RawTherapee/files/573454/rawtherapeelog.txt)

gdb.exe crashed and stopped logging, got the "gdb.exe has stopped working" message, but there's what it was able to log.

**Comment by Floessie** [comment]
Hm, not really in the HaldCLUT code, nor in the commit @Beep6581 mentioned, or do I miss something? I wish there was a way to reproduce it here. And you are surely not running out of memory? Does changing the CLUT cache size in the preferences help?

**Comment by torridgristle** [comment]
Changing HaldCLUT cache size from 4 to 8 doesn't prevent it from crashing, and regarding memory Task Manager says Physical Memory 51%: Total 6134, cached 2884, available 2872, free 60, and when I close RawTherapee it changes to Physical Memory 42%: Total 6134, Cached 2937, available 3499, free 624.

**Comment by Floessie** [comment]
Cannot reproduce it here under Linux.

@heckflosse There is an interesting comment [here](https://github.com/Beep6581/RawTherapee/blob/master/rtengine/imagefloat.cc#L479). Is it still valid and could it have something to do with @torridgristle's problem?

**Comment by heckflosse** [comment]
@Floessie If I'm not totally wrong, the per thread memory allocation is already done there and we should correct the comment to avoid confusion ;-)

**Comment by Floessie** [comment]
@heckflosse I see, didn't know which allocation was meant. So we're still in the dark...

**Comment by torridgristle** [comment]
I was able to add some more HaldCLUTs to RawTherapee by adding a new directory "Faded and Moody" to a pre-existing directory "VSCO Cam" which is located in "C:\Program Files\RawTherapee\HaldCLUT\VSCO Cam\Faded and Moody".

I thought that this might mean that I can add the previous "Mine" directory with the 1 HaldCLUT back in if it's inside another directory so I tried placing "Mine" into the pre-existing "Color" which is located in "C:\Program Files\RawTherapee\HaldCLUT\Color\Mine" but it crashed when I clicked on the Film Emulation dropdown.

Here's one of the Faded and Moody HaldCLUTs, they're all the same size and format.
![t1](https://cloud.githubusercontent.com/assets/23273466/20039626/17329dc2-a415-11e6-8f58-066a69b707c6.png)

**Comment by Beep6581** [comment]
@torridgristle can you still reproduce this in 5.5(rc*)?
